### PR TITLE
Make recorded Timeline streams configurable from DevTools dialog.

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This PR also moves the Network Traffic toggle into the dialog and enables it on the Timeline by default.

Screen Shot 2020-05-11 at 2 55 45 PM
Screen Shot 2020-05-11 at 2 54 40 PM

### CODE REVIEW

File: packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart [hunk 1]
```diff
@@ -229,30 +228,25 @@ class TimelineScreenBodyState extends State<TimelineScreenBody>
...
+        const SizedBox(width: defaultSpacing),
+        ActionButton(
+          child: OutlineButton(
+            child: const Icon(
+              Icons.more_vert,
+              size: defaultIconSize,
+            ),
+            onPressed: _openSettingsDialog,
+          ),
+          tooltip: 'More Timeline Configurations',
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart:240]: `Timeline  Configuration`

File: packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart [hunk 2]
```diff
@@ -333,3 +327,79 @@ class TimelineScreenBodyState extends State<TimelineScreenBody>
...
+          mainAxisAlignment: MainAxisAlignment.start,
+          children: [
+            checkbox,
+            title,
+          ],
+        ),
+      );
+    }
+
+    // Special case "Network Traffic" because it is not implemented as a
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart:384]: If this does have some differences from the POV of the user, we could use a `Divider` widget to visually separate it from the other items.

File: packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart [hunk 3]
```diff
@@ -333,3 +327,79 @@ class TimelineScreenBodyState extends State<TimelineScreenBody>
...
+  }
+
+  List<Widget> _recordedStreams() {
+    final settings = <Widget>[];
+    for (final streamName in controller.recordedStreams.keys) {
+      final title = Text(streamName);
+      final checkbox = ValueListenableBuilder(
+        valueListenable: controller.recordedStreams[streamName],
+        builder: (context, enabled, _) {
+          return Checkbox(
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart:366]: For many of these items, the user may not have a sense for what toggling them does. For the ones where we can come up with good, brief descriptions, perhaps include the description in-line and to the right pg the checkbox, w/ a subtle text style?

File: packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart:384]: I thought about that. I don't think it does have a difference from the POV of the user. To the user, all the other toggles mean "do I want to see these events in the timeline or not", and this is what the result of the Network toggle will be for the user as well.

File: packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart (hunk 3 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart:366]: That's a good idea. Added a TODO as I'm following up with @bkonyi to find accurate docs on these.

Comment (devoncarew) [reply packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart:367]: OK; we may just need a good enough description to let the user know why they want want to turn these on.

I assume for many of them they really won't want to touch them. Some they will know what they do and want to enable them, like the network toggle. Not sure if there's a way to call out the two types well in the UI - have a regular and an advanced section?

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart:367]: The observatory categorizes different sets like so:
Screen Shot 2020-05-12 at 7 52 12 AM

Clicking one of these profiles enables a different set of timeline flags

**Pull request merged**